### PR TITLE
Github actions CI migration

### PR DESCRIPTION
**Pull request by djcp**
This PR can serve as an example for using github actions CI for our public open source projects. **We specifically should ensure that all external contributors require approval to run workflows to ensure we are careful about who can run code in the ezcater context.**

## What did we change?

- Build on multiple more modern rubies
- Bump up Bundler
- Remove old CI vendor configurations
- Update to use github actions CI

## Why are we doing this?

To migrate to github actions CI rather than other "legacy" vendors

## How was it tested?
- [x] Specs
- [ ] Locally
- [ ] Staging

### CODE REVIEW

**Comment by djcp** [comment]
@corymurphy @tuckerweibell I pinged you on this PR as it's the first public repo I'm migrating to github actions and I wanted to make sure we're OK from a security standpoint.

**Comment by djcp** [comment]
This should pass once we remove the circleci webhook

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-ruby-2.5.1
+ruby-3.2.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.22.2
```
Comment (djcp) [inline .ruby-version]: lol. vimfail

**Comment by corymurphy** [comment]
> @corymurphy @tuckerweibell I pinged you on this PR as it's the first public repo I'm migrating to github actions and I wanted to make sure we're OK from a security standpoint.

I'm cool with it, do we have this repo restricted such that you need to create a pull request from an org fork instead of a branch within this repo? Just trying to avoid someone using our compute resources for CI if they want to contribute.

**Comment by djcp** [comment]
@corymurphy @tuckerweibell we require approval for all outside collaborators under github actions settings. I BELIEVE this will mean all non-ezcater accounts that fork will need to be approved before actions can be run against their contributions, no exceptions.

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

and the current setting:

image

I think this is a good default to start until we have clearer guidance based on @tuckerweibell 's upcoming investigations.

**Review by mikowitz — APPROVED** [review]

**Pull request merged**